src/forklift/arcgis.py

```diff
@@ -114,12 +114,18 @@ def _fetch(self, url):
         if self.token_expire_milliseconds <= time() * 1000:
             self._request_token()
 
+        ok = False
         data = {'f': 'json', 'token': self.token}
-
-        r = requests.post(url, data=data)
-        r.raise_for_status()
-
-        ok = self._return_false_for_status(r.json())
+        
+        try:
+            r = requests.post(url, data=data, timeout=30)
+            r.raise_for_status()
+
+            ok = self._return_false_for_status(r.json())
+        except requests.exceptions.Timeout:
+            return False
```

Comment:
This would skip the sleep line below. Not sure how important that is but maybe it should be changed to `ok = False`.

Reply:
Would that matter if it took 30 seconds to timeout?

Reply:
Good point.